lib/livereload.coffee

```diff
@@ -78,8 +78,14 @@ class Server
     exts = @config.exts
     fileext = path.extname filepath
                   .substring 1
+
     for ext in exts when ext == fileext
-      @refresh filepath
+      delayedRefresh = setTimeout(
+        =>
+          clearTimeout(delayedRefresh)
+          @refresh filepath
+        @config.delay
```

Comment:
I have questions about this routine. Even if there is no delay specified, this is still going to create and drop a timeout. That seems wasteful. Probably should check if delay is 0?

Also, this creates a timeout for each file. If I'm watching 25 files, will this create 25 delays? I'm shaky on how that works.